### ISSUE TITLE
enw_preprocess_data throws error in presence of zero counts

### ISSUE THREAD

**Issue by rumackaaron** [epinowcast/epinowcast#541]
**Describe the bug**
In some cases, when calling `enw_preprocess_data` on data that has zero counts, the function can throw the following error:

> Error in if (warn && mean_coverage > 0.5) { :
  missing value where TRUE/FALSE needed

The traceback is as follows:
> 2: check_max_delay(out[], warn = TRUE, warn_internal = TRUE)
1: enw_preprocess_data(incidence, max_delay = 4, by = "site")

**To Reproduce**
1. Load the [attached csv file](https://github.com/user-attachments/files/17799903/enw_preprocess_bug_data.csv) as `incidence`
2. Call `enw_preprocess_data(incidence, max_delay = 4, by ="site")`

**Expected behavior**
I expected the function to return a preprocessed object as specified by the `enw_preprocess_data` documentation.

**Desktop (please complete the following information):**
 - R version 4.4.1
 - Epinowcast version 0.3.0.1000

**Additional context**
The error might be related to the `by` argument. If I call `enw_preprocess_data` on the counts of either individual site, I don't get an error.

**Comment by seabbs** [comment]
@rumackaaron thanks for the report. @adrian-lison this looks to be coming from the ad-hoc checks that the delay is long enough. @rumackaaron can you confirm that some sites in your data are all 0 counts as this seems like it could be driving this?

**Comment by rumackaaron** [comment]
> @rumackaaron can you confirm that some sites in your data are all 0 counts as this seems like it could be driving this?

In the example I gave, both sites have at least 1 count, although most reference dates have 0 counts. I do think that the 0 counts are driving this, but the bug occurs even when sites don't have all 0 counts.

**Comment by seabbs** [comment]
they have non zero counts within the max delay you set or just in general? If you have any luck finding the subset causing problems that would be really great

**Comment by rumackaaron** [comment]
Yes, in the attached csv, there are nonzero counts within the max delay. I tried without success to find the exact subset which causes the error

**Comment by seabbs** [comment]
Sorry to be clear I am asking if for any site there are are no non-zero elements within the max delay

**Comment by rumackaaron** [comment]
Each site has non-zero elements within the max delay

**Comment by seabbs** [comment]
Awesome thanks for checking

**Comment by seabbs** [comment]
<img width="265" alt="Screenshot 2024-11-19 at 20 54 33" src="https://github.com/user-attachments/assets/050a5ef3-7073-4b0d-9536-f1664a8861f9">

Is it both A and B that give you problems here? My expectation is that there only being a single count in A and that not being reported out to the max delay is going to fail (its trying to take the mean coverage of nothing). If B also fails then that is interesting and suggests the current implementation can't handle completely 0 report reference pairs which you have here

**Comment by rumackaaron** [comment]
> The error might be related to the `by` argument. If I call `enw_preprocess_data` on the counts of either individual site, I don't get an error.

When I filter for site A and run `enw_preprocess_data` without `by = 'site'`, the code runs smoothly, but the same is true for filtering for site B and running without `by = 'site'`

**Comment by seabbs** [comment]
Oh okay weird interesting. I'll take a look asap
